bigquery/google/cloud/bigquery/job.py

```diff
@@ -1298,6 +1298,84 @@ def query_plan(self):
         plan_entries = self._job_statistics().get('queryPlan', ())
         return [copy.deepcopy(entry) for entry in plan_entries]
 
+    @property
+    def total_bytes_processed(self):
+        """Return total bytes processed from job statistics, if present.
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#statistics.query.totalBytesProcessed
+
+        :rtype: int or None
+        :returns: total bytes processed by the job, or None if job is not
+                  yet complete.
+        """
+        return self._job_statistics().get('totalBytesProcessed')
```

Comment:
totalBytesProcessed comes back as a string, actually. So the client lib needs to convert it to an int.

(Reason it is a string is JSON assumes double for numeric values, so sent as string to avoid losing precision)